qiskit_nature/second_q/mappers/qubit_mapper.py

```diff
@@ -61,6 +104,49 @@ def map(self, second_q_op: SparseLabelOp) -> PauliSumOp:
         """
         raise NotImplementedError()
 
+    def map_all(
```

Comment:
I am still pondering the necessity of `map` *and* `map_all` to exist. Since `map_all` can also handle a single operator, it begs the question whether we need to differentiate these or whether we can simply extend the functionality of `map` to also support `ListOrDictType`. :thinking:

Reply:
I have replaced all local definition of `map()` to a private `_map_single()` and then the `qubitmapper()` class implements the `map()` method accepting a `SparseLabelOp | ListOrDictType[SparseLabelOp]` and returning a `PauliSumOp | ListOrDictType[PauliSumOp]`